src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -839,13 +861,16 @@ private SecurityStatusPal GenerateToken(byte[] input, int offset, int count, ref
                     //
                     if (!cachedCreds && _securityContext != null && !_securityContext.IsInvalid && _credentialsHandle != null && !_credentialsHandle.IsInvalid)
                     {
-                        SslSessionsCache.CacheCredential(_credentialsHandle, thumbPrint, _sslProtocols, _serverMode, _encryptionPolicy);
+                        SslSessionsCache.CacheCredential(_credentialsHandle, thumbPrint, _sslAuthenticationOptions.EnabledSslProtocols, _sslAuthenticationOptions.IsServer, _sslAuthenticationOptions.EncryptionPolicy);
                     }
                 }
             }
 
             output = outgoingSecurity.token;
-            
+
+            byte[] alpnResult = SslStreamPal.GetNegotiatedApplicationProtocol(_securityContext);
+            _negotiatedApplicationProtocol = alpnResult == null ? default : new SslApplicationProtocol(alpnResult, false);
```

Comment:
Is default a variable I'm not seeing?  Or is there a new C# language feature to just have the value `default` which means `default(ImplicitT)`?

Reply:
Yes, it's new in C# 7.1